src/bun.js/events.exports.js

```diff
@@ -0,0 +1,464 @@
+// Reimplementation of https://nodejs.org/api/events.html
+// Reference: https://github.com/nodejs/node/blob/main/lib/events.js
+var { isPromise, Array, Object } = import.meta.primordials;
+const SymbolFor = Symbol.for;
+const ObjectDefineProperty = Object.defineProperty;
+const kCapture = Symbol("kCapture");
+const kErrorMonitor = SymbolFor("events.errorMonitor");
+const kMaxEventTargetListeners = Symbol("events.maxEventTargetListeners");
+const kMaxEventTargetListenersWarned = Symbol("events.maxEventTargetListenersWarned");
+const kWatermarkData = SymbolFor("nodejs.watermarkData");
+const kRejection = SymbolFor("nodejs.rejection");
+const captureRejectionSymbol = SymbolFor("nodejs.rejection");
+const ArrayPrototypeSlice = Array.prototype.slice;
+
+var defaultMaxListeners = 10;
+
+// EventEmitter must be a standard function because some old code will do weird tricks like `EventEmitter.apply(this)`.
+function EventEmitter(opts) {
+  if (this._events === undefined || this._events === this.__proto__._events) {
+    this._events = { __proto__: null };
+    this._eventsCount = 0;
+  }
+
+  this._maxListeners ??= undefined;
+  if (
+    (this[kCapture] = opts?.captureRejections ? Boolean(opts?.captureRejections) : EventEmitter.prototype[kCapture])
+  ) {
+    this.emit = emitWithRejectionCapture;
+  }
+}
+
+EventEmitter.prototype._events = undefined;
```

Comment:
instead of 

```js
EventEmitter.prototype
```

People might mess with it at runtime

Instead, lets assign it to a local variable and use that one

```js
const EventEmitterPrototype = EventEmitter.prototype;
```